app/src/main/java/bg/zahov/app/ui/exercise/info/ExerciseInfoViewModel.kt

```diff
@@ -0,0 +1,86 @@
+package bg.zahov.app.ui.exercise.info
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import bg.zahov.app.Inject
+import bg.zahov.app.data.interfaces.WorkoutProvider
+import bg.zahov.app.data.model.state.ExerciseHistoryData
+import com.github.mikephil.charting.data.Entry
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+
+/**
+ * ViewModel class for managing the UI state and business logic of the Exercise Info screen.
+ *
+ * @property workoutProvider The provider that fetches exercise history data. By default, it uses the injected instance from [Inject.workoutProvider].
+ */
+class ExerciseInfoViewModel(private val workoutProvider: WorkoutProvider = Inject.workoutProvider) :
+    ViewModel() {
+    //Internal state flow for managing UI data updates.
+    private val _uiState = MutableStateFlow(ExerciseHistoryData())
+
+    /**
+     * Public state flow that exposes the UI state to observers.
+     */
+    val uiState: StateFlow<ExerciseHistoryData> = _uiState
+
+    /**
+     * Initializes the ViewModel and begins collecting exercise history data from the [WorkoutProvider].
+     */
+    init {
+        viewModelScope.launch {
+            workoutProvider.getExerciseHistory().collect { data ->
```

Comment:
extract it in different function - getExercisesHistory for example, also you can extract mapping logic in different functions as well for better readability

Reply:
fixed